server/src/test/java/org/elasticsearch/rest/action/document/RestBulkActionTests.java

```diff
@@ -45,32 +46,38 @@
 public class RestBulkActionTests extends ESTestCase {
 
     public void testBulkPipelineUpsert() throws Exception {
-        final NodeClient mockClient = mock(NodeClient.class);
-        final Map<String, String> params = new HashMap<>();
-        params.put("pipeline", "timestamps");
-        new RestBulkAction(settings(Version.CURRENT).build())
-            .handleRequest(
-                new FakeRestRequest.Builder(
-                    xContentRegistry()).withPath("my_index/_bulk").withParams(params)
-                    .withContent(
-                        new BytesArray(
-                            "{\"index\":{\"_id\":\"1\"}}\n" +
-                                "{\"field1\":\"val1\"}\n" +
-                                "{\"update\":{\"_id\":\"2\"}}\n" +
-                                "{\"script\":{\"source\":\"ctx._source.counter++;\"},\"upsert\":{\"field1\":\"upserted_val\"}}\n"
-                        ),
-                        XContentType.JSON
-                    ).withMethod(RestRequest.Method.POST).build(),
-                mock(RestChannel.class), mockClient
-            );
-        Mockito.verify(mockClient)
```

Comment:
By removing this, we no longer verify that the bulk method is actually called. I think adding an AtomicInteger as a counter for method invocations and validating that the method is called once should be sufficient

Reply:
++, I did that for a number of other tests that needed similar change, but forgot on this one.